Index: linux-3.10.44/drivers/net/ethernet/marvell/mvmdio.c
===================================================================
--- linux-3.10.44.orig/drivers/net/ethernet/marvell/mvmdio.c
+++ linux-3.10.44/drivers/net/ethernet/marvell/mvmdio.c
@@ -138,7 +138,12 @@ static int orion_mdio_read(struct mii_bu
 
 	mutex_unlock(&dev->lock);
 
-	return val & 0xFFFF;
+	if(mii_id == 0x10 && (regnum == 0x02 || regnum == 0x03)) {
+		pr_info("mvmdio: faking ID register value\n");
+		return 0x1234;
+	} else {
+		return val & 0xFFFF;
+	}
 }
 
 static int orion_mdio_write(struct mii_bus *bus, int mii_id,
