--- a/drivers/net/wireless/realtek/rtlwifi/efuse.c
+++ b/drivers/net/wireless/realtek/rtlwifi/efuse.c
@@ -1274,7 +1274,7 @@ int rtl_get_hwinfo(struct ieee80211_hw *
 
 	rtlefuse->eeprom_channelplan = *&hwinfo[params[6]];
 	rtlefuse->eeprom_version = *(u16 *)&hwinfo[params[7]];
-	rtlefuse->txpwr_fromeprom = true;
+	rtlefuse->txpwr_fromeprom = false;
 	rtlefuse->eeprom_oemid = *&hwinfo[params[8]];
 
 	rtl_dbg(rtlpriv, COMP_INIT, DBG_LOUD,
