--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -3530,14 +3530,14 @@ static struct phy_driver marvell_drivers
 		.phy_id_mask = MARVELL_PHY_ID_MASK,
 		.name = "Marvell 88E1540",
 		.driver_data = DEF_MARVELL_HWMON_OPS(m88e1510_hwmon_ops),
-		/* PHY_GBIT_FEATURES */
+		/* PHY_GBIT_FEATURES irq disabled */
 		.flags = PHY_POLL_CABLE_TEST,
 		.probe = marvell_probe,
 		.config_init = marvell_1011gbe_config_init,
 		.config_aneg = m88e1510_config_aneg,
 		.read_status = marvell_read_status,
-		.config_intr = marvell_config_intr,
-		.handle_interrupt = marvell_handle_interrupt,
+//		.config_intr = marvell_config_intr,
+//		.handle_interrupt = marvell_handle_interrupt,
 		.resume = genphy_resume,
 		.suspend = genphy_suspend,
 		.read_page = marvell_read_page,
