--- a/include/linux/marvell_phy.h
+++ b/include/linux/marvell_phy.h
@@ -18,7 +18,7 @@
 #define MARVELL_PHY_ID_88E1340S		0x01410dc0
 #define MARVELL_PHY_ID_88E1116R		0x01410e40
 #define MARVELL_PHY_ID_88E1510		0x01410dd0
-#define MARVELL_PHY_ID_88E1540		0x01410eb0
+#define MARVELL_PHY_ID_88E1540		0x01410000/*break this manually because this driver is incompatible with MVKW2 internal GE-PHY*/
 #define MARVELL_PHY_ID_88E1545		0x01410ea0
 #define MARVELL_PHY_ID_88E1548P		0x01410ec0
 #define MARVELL_PHY_ID_88E3016		0x01410e60
